src/CoreWCF.Http/tests/MessageEncodingTests.cs

```diff
@@ -189,5 +203,51 @@ protected override Channels.Binding GetServerBinding()
                 return binding;
             }
         }
+
+        internal class AssertMtomOptimizedEndpointBehvaior : System.ServiceModel.Description.IEndpointBehavior
+        {
+            class AssertMtomOptimizedDelegatingHandler : DelegatingHandler
```

Comment:
I first tried to use a `IClientMessageInspector.AfterReceiveReply` to inspect the response client side but the provided Message was already 'rebuilt' hiding the multipart.